src/core/ComboBox/ComboBox.tsx

```diff
@@ -103,25 +119,44 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     [options, id],
   );
 
+  const userOnChange = React.useRef(onChange);
```

Comment:
What was the part that you said caused too much rerenders? Was it this one?

Reply:
Yeah, this combined with the actual returned menu items, which I'm now memoizing right below this (previously I wasn't).